opt/scopeserver/dataserver/utils/constant.py

```diff
@@ -273,9 +274,17 @@ class Species(Enum):
 ]
 
 
-def to_colours(index: Iterable[int]) -> List[str]:
+def to_colours(index: Iterable[int], color_list=None) -> List[str]:
     """Convert indexes (`index`) into pre-determined HTML hex colour values.
     `index` can be a `range(some_number)` or any iterable collection of numbers.
     """
-    num_colours = len(BIG_COLOR_LIST)
-    return [BIG_COLOR_LIST[i % num_colours] for i in index]
+    if color_list is not None and len(np.unique(index)) > len(color_list):
+        raise Exception(f"Not enough custom colors defined.")
```

Comment:
I would rather this report the error and then fallback to using the `BIG_COLOR_LIST`. This way, something is at least returned.

Reply:
Fixed using @KrisDavie suggestion:
```
def to_colours(index: Iterable[int], color_list=None) -> List[str]:
    """Convert indexes (`index`) into pre-determined HTML hex colour values.
    `index` can be a `range(some_number)` or any iterable collection of numbers.
    """

    if color_list is None:
        final_color_list = BIG_COLOR_LIST
    elif len(np.unique(index)) > len(color_list):
        logger.warning(f"Not enough custom colors defined. Falling back to BIG_COLOR_LIST")
        final_color_list = BIG_COLOR_LIST
    else:
        final_color_list = [color[1:] if color.startswith("#") else color for color in color_list]

    num_colours = len(final_color_list)

    return [final_color_list[i % num_colours] for i in index]
```